src/shared/codelens/csharpCodeLensProvider.ts

```diff
@@ -380,3 +400,60 @@ export function isPublicMethodSymbol(
 export function generateDotNetLambdaHandler(components: DotNetLambdaHandlerComponents): string {
     return `${components.assembly}::${components.namespace}.${components.class}::${components.method}`
 }
+
+export async function installDebugger({
+    runtime,
+    codeUri
+}: {
+    runtime: string,
+    codeUri: string
+}): Promise<string> {
+    const vsdbgPath = path.resolve(codeUri, '.vsdbg')
+
+    try {
+        await access(vsdbgPath)
+
+         // vsdbg is already installed.
+        return vsdbgPath
+    } catch {
+        // We could not access vsdbgPath. Swallow error and continue.
+    }
+
+    try {
+        await mkdir(vsdbgPath)
+
+        const process = crossSpawn(
+            'docker',
```

Comment:
Internal discussions for https://github.com/aws/aws-toolkit-vscode/issues/376 concluded that we should have some sort of check to see if Docker images needed to be updated, which would require a separate call to Docker.

Should we instead create wrapper functions for Docker calls?

Reply:
For this PR, I've moved communication with docker to a simple `DockerClient` interface, with a TODO to adopt `docker-node-api` or similar. I agree that we need to address validation in a separate PR.